src/fhir/utilities.ts

```diff
@@ -0,0 +1,177 @@
+        break;
+      case 'ValueSet':
+        collectionString = `${constants.COLLECTION.VALUESET}_${baseVersion}`;
+        historyCollectionString = `${constants.COLLECTION.VALUESET}_${baseVersion}_History`;
+        break;
+    }
+
+    const Resource = resolveSchema(baseVersion, resource.resourceType);
+    const fhirResource = new Resource(resource);
+
+    // Create the resource's metadata
+    const Meta = FhirUtilities.getMeta(baseVersion);
+    fhirResource.meta = new Meta({
+      versionId: '1',
+      lastUpdated: moment.utc().format('YYYY-MM-DDTHH:mm:ssZ')
+    });
+
+    if (collectionString === '') {
+      return reject('    Unsupported FHIR Resource Type');
+    }
+    const collection = db.collection(collectionString);
+
+    // Create the document to be inserted into teh database
+    const doc = JSON.parse(JSON.stringify(fhirResource.toJSON()));
+    Object.assign(doc, { id: id });
+
+    // Create a clone of the object without the _id parameter before assigning a value to
+    // the _id parameter in the original document
+    const history_doc = Object.assign({}, doc);
+    Object.assign(doc, { _id: id });
+
+    // Insert our resource record
+    collection.insert(doc, (err: any) => {
+      if (err) {
+        console.log('    Error with %s.create: ', resource.resourceType, err.message);
+        return reject(err);
+      } else {
+        console.log('    Successfully added ' + resource.resourceType + ' -- ' + id);
+      }
+
+      // Save the resource to history
+      const history_collection = db.collection(historyCollectionString);
+
+      // Insert our patient record to history but don't assign _id
+      return history_collection.insert(history_doc, (err2: any) => {
+        if (err2) {
+          console.log('    Error with %sHistory.create: ', resource.resourceType, err2.message);
+          return reject(err2);
+        }
+        return resolve({ id: doc.id, resource_version: doc.meta.versionId });
+      });
+    });
+  }
+
+  static loadResources(resourcePath: string) {
+    console.log('Loading FHIR Resources from: ' + resourcePath);
+
+    // Loop through all the files in the temp directory
+    fs.readdir(resourcePath, function (err: any, files: any) {
+      if (err) {
+        console.error('Could not list the directory.', err);
+        process.exit(1);
+      }
+
+      files.forEach(function (file: any) {
+        // Make one pass and make the file complete
+        const filePath = path.join(resourcePath, file);
+
+        fs.stat(filePath, function (error: any, stat: any) {
+          if (error) {
+            console.error('Error getting file statistics.', error);
+            return;
+          }
+
+          if (stat.isFile()) {
+            const extension = re.exec(filePath);
+            if (extension) {
+              if (extension[1].toLowerCase() === 'json') {
+                if (file !== 'TopicMetadata.json') {
```

Comment:
I think this is causing a failure on my end when trying to load resources by marking an empty json file (TIRF prepop file) as a resource file and trying to load it.  We could validate the resource before loading it, or we could just add a try/except and make note of the failed load in the log.

As a separate issue, my understanding is that this is supposed to load the FHIR resources, which is being ruled in here as `json files that are not named TopicMetadata.json` when it should really be `files located in the "resources" directory of a rule`.  This way it's less ambiguous about which files belong in the FHIR server.

Reply:
I'm adding an error that'll go to the log on a file that is not able to be parsed or is not a valid FHIR resource. I was hesitant to build in any specifics of the folder structure since I thought we may go away from the structure we have now at some point. The rules don't really exist as they did for CRD since we have them hard coded into the REMS admin for now.
I'll add in the folder structure for now along with the error handling.